mlflow/tracking/client.py

```diff
@@ -905,22 +907,41 @@ def _validate_prompt(self, name: str, version: int):
 
     def parse_prompt_uri(self, uri: str) -> tuple[str, str]:
         """
-        Parse prompt URI into prompt name and prompt version.
-        - 'prompt:/<name>/<version>' -> ('<name>', '<version>')
-        - 'prompt:/<name>@<alias>' -> ('<name>', '<version>')
+        Parse prompt URI into prompt name and prompt version or alias.
+        - 'prompts:/<name>/<version>' -> ('<name>', '<version>')
+        - 'prompts:/<name>@<alias>' -> ('<name>', '<alias>')
         """
         parsed = urllib.parse.urlparse(uri)
 
         if parsed.scheme != "prompts":
             raise MlflowException.invalid_parameter_value(
-                f"Invalid prompt URI: {uri}. Expected schema 'prompts:/<name>/<version>'"
+                f"Invalid prompt URI: {uri}. "
+                "Expected schema 'prompts:/<name>/<version>' or 'prompts:/<name>@<alias>'"
             )
 
-        # Replace schema to 'models:/' to reuse the model URI parsing logic
-        try:
-            return get_model_name_and_version(self, f"models:{parsed.path}")
-        except MlflowException:
-            raise MlflowException(f"Prompt '{uri}' does not exist.", RESOURCE_DOES_NOT_EXIST)
+        path = parsed.path.lstrip("/")
```

Comment:
I think there are existing utilities for extracting this info for model versions. E.g. they should be using `urlparse` under the hood. Can we use those? If not, we should at least be using urlparse